addons/sale/models/sale_order.py

```diff
@@ -806,29 +806,52 @@ def action_unlock(self):
         self.write({'state': 'sale'})
 
     def action_cancel(self):
+        """ Cancel SO after showing the cancel wizard when needed. (cfr `_show_cancel_wizard`)
+
+        For post-cancel operations, please only override `_action_cancel`.
+
+        note: self.ensure_one() if the wizard is shown.
+        """
         cancel_warning = self._show_cancel_wizard()
         if cancel_warning:
+            self.ensure_one()
+            template_id = self.env['ir.model.data']._xmlid_to_res_id(
+                'sale.mail_template_sale_cancellation', raise_if_not_found=False
```

Comment:
Does the following code even work if the template is deleted ?
(If yes, keep like that, if no, use a simple ref to make sure the record exists)

Reply:
Just tested it, it works :partying_face: